// @kiwicom/orbit-components/lib/Table/TableBody/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledTableBody = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledTableBody = (0, _styledComponents.default)(function (_ref) {
  var children = _ref.children,
      className = _ref.className;
  return React.createElement("tbody", {
    className: className
  }, children);
}).withConfig({
  displayName: "TableBody__StyledTableBody",
  componentId: "sc-1uovf8k-0"
})(["width:100%;white-space:nowrap;border-left:1px solid ", ";border-right:1px solid ", ";border-bottom:1px solid ", ";"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.borderColorTable;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.borderColorTable;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.orbit.borderColorTable;
});
exports.StyledTableBody = StyledTableBody;
StyledTableBody.defaultProps = {
  theme: _defaultTheme.default
};

var TableBody = function TableBody(props) {
  return React.createElement(StyledTableBody, null, props.children);
};

var _default = TableBody;
exports.default = _default;
