// @kiwicom/orbit-components/lib/ChoiceGroup/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Heading = _interopRequireDefault(require("../Heading"));

var _Stack = _interopRequireDefault(require("../Stack"));

var _consts = require("./consts");

var _FormFeedback = _interopRequireWildcard(require("../FormFeedback"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _FilterWrapper = _interopRequireDefault(require("./components/FilterWrapper"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var getHeadingSize = function getHeadingSize(size) {
  var _SIZES;

  var SIZES = (_SIZES = {}, _defineProperty(_SIZES, _consts.LABEL_SIZES.NORMAL, "title3"), _defineProperty(_SIZES, _consts.LABEL_SIZES.LARGE, "title2"), _SIZES);
  return SIZES[size];
};

var StyledChoiceGroup = _styledComponents.default.div.withConfig({
  displayName: "ChoiceGroup__StyledChoiceGroup",
  componentId: "njp29m-0"
})(["width:100%;display:flex;flex-direction:column;", "{position:relative;margin-top:", ";top:initial;}"], _FormFeedback.StyledFormFeedback, function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.spaceXSmall;
});

StyledChoiceGroup.defaultProps = {
  theme: _defaultTheme.default
};

var ChoiceGroup =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(ChoiceGroup, _React$PureComponent);

  function ChoiceGroup() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, ChoiceGroup);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(ChoiceGroup)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "handleChange", function (ev) {
      ev.persist();
      var onChange = _this.props.onChange;

      if (onChange) {
        onChange(ev);
      }
    });

    _defineProperty(_assertThisInitialized(_this), "handleOnly", function (ev, child) {
      if (_this.props.onOnlySelection) {
        _this.props.onOnlySelection(ev, child);
      }
    });

    return _this;
  }

  _createClass(ChoiceGroup, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          dataTest = _this$props.dataTest,
          label = _this$props.label,
          _this$props$labelSize = _this$props.labelSize,
          labelSize = _this$props$labelSize === void 0 ? _consts.LABEL_SIZES.NORMAL : _this$props$labelSize,
          _this$props$labelElem = _this$props.labelElement,
          labelElement = _this$props$labelElem === void 0 ? _consts.LABEL_ELEMENTS.H4 : _this$props$labelElem,
          error = _this$props.error,
          children = _this$props.children,
          filter = _this$props.filter,
          onOnlySelection = _this$props.onOnlySelection;
      return React.createElement(StyledChoiceGroup, {
        "data-test": dataTest
      }, label && React.createElement(_Heading.default, {
        type: getHeadingSize(labelSize),
        element: labelElement,
        spaceAfter: "medium"
      }, label), React.createElement(_Stack.default, {
        direction: "column",
        spacing: filter ? "none" : "condensed"
      }, React.Children.map(children, function (child) {
        return !filter ? React.createElement(React.Fragment, null, React.cloneElement(child, {
          onChange: _this2.handleChange,
          hasError: !!error
        })) : React.createElement(_FilterWrapper.default, {
          child: child,
          onClick: _this2.handleOnly,
          onOnlySelection: onOnlySelection
        }, React.createElement(React.Fragment, null, React.cloneElement(child, {
          onChange: _this2.handleChange,
          hasError: !!error
        })));
      })), error && React.createElement(_FormFeedback.default, {
        type: "error",
        fixed: true
      }, error));
    }
  }]);

  return ChoiceGroup;
}(React.PureComponent);

var _default = ChoiceGroup;
exports.default = _default;
