// @kiwicom/orbit-components/lib/Breadcrumbs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BreadcrumbsItem", {
  enumerable: true,
  get: function get() {
    return _BreadcrumbsItem.default;
  }
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _Button = _interopRequireDefault(require("../Button"));

var _ChevronLeft = _interopRequireDefault(require("../icons/ChevronLeft"));

var _BreadcrumbsItem = _interopRequireDefault(require("./BreadcrumbsItem"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledBreadcrumbs = _styledComponents.default.nav.withConfig({
  displayName: "Breadcrumbs__StyledBreadcrumbs",
  componentId: "sc-1zqkwq-0"
})(["font-family:", ";font-size:", ";"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.fontFamily;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.fontSizeTextSmall;
});

StyledBreadcrumbs.defaultProps = {
  theme: _defaultTheme.default
};

var StyledBreadcrumbsList = _styledComponents.default.ol.withConfig({
  displayName: "Breadcrumbs__StyledBreadcrumbsList",
  componentId: "sc-1zqkwq-1"
})(["display:flex;flex-wrap:wrap;list-style:none;margin:0;padding:0;"]);

var StyledBackButtonWrapper = _styledComponents.default.span.withConfig({
  displayName: "Breadcrumbs__StyledBackButtonWrapper",
  componentId: "sc-1zqkwq-2"
})(["margin-right:", ";"], function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.spaceSmall;
});

StyledBackButtonWrapper.defaultProps = {
  theme: _defaultTheme.default
};

var Breadcrumbs = function Breadcrumbs(_ref4) {
  var children = _ref4.children,
      dataTest = _ref4.dataTest,
      onGoBack = _ref4.onGoBack;
  return React.createElement(StyledBreadcrumbs, {
    "aria-label": "Breadcrumb",
    role: "navigation",
    "data-test": dataTest
  }, React.createElement(StyledBreadcrumbsList, {
    vocab: "http://schema.org/",
    typeof: "BreadcrumbList"
  }, onGoBack && React.createElement(StyledBackButtonWrapper, null, React.createElement(_Button.default, {
    iconLeft: React.createElement(_ChevronLeft.default, null),
    circled: true,
    type: "secondary",
    size: "small",
    onClick: onGoBack
  })), React.Children.map(children, function (item, key) {
    return React.cloneElement(item, {
      active: key === React.Children.count(children) - 1,
      contentKey: key + 1
    });
  })));
};

var _default = Breadcrumbs;
exports.default = _default;
