// @kiwicom/orbit-components/lib/Modal/ModalContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withModalContext = exports.ModalContext = void 0;

var React = _interopRequireWildcard(require("react"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var ModalContext = React.createContext({
  setDimensions: function setDimensions() {},
  decideFixedFooter: function decideFixedFooter() {},
  setHasModalSection: function setHasModalSection() {},
  removeHasModalSection: function removeHasModalSection() {},
  manageFocus: function manageFocus() {},
  hasModalSection: false,
  isMobileFullPage: false
});
exports.ModalContext = ModalContext;

var withModalContext = function withModalContext(Component) {
  return function (props) {
    return React.createElement(ModalContext.Consumer, null, function (contextProps) {
      return React.createElement(Component, _extends({}, props, contextProps));
    });
  };
};

exports.withModalContext = withModalContext;
