// @kiwicom/orbit-components/lib/TripSector/TripDate/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Calendar = _interopRequireDefault(require("../../icons/Calendar"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _Text = _interopRequireWildcard(require("../../Text"));

var _Clock = _interopRequireDefault(require("../../icons/Clock"));

var _Stack = _interopRequireDefault(require("../../Stack"));

var _Badge = _interopRequireDefault(require("../../Badge"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledTripDate = _styledComponents.default.div.withConfig({
  displayName: "TripDate__StyledTripDate",
  componentId: "sc-1t2zo5b-0"
})(["display:flex;align-items:center;margin-bottom:", ";padding-bottom:", ";background:", ";z-index:1;", "{font-weight:", ";height:", ";}"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.spaceXSmall;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.spaceXXSmall;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.paletteWhite;
}, _Text.StyledText, function (_ref4) {
  var theme = _ref4.theme;
  return theme.orbit.fontWeightMedium;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.orbit.heightIconSmall;
});

StyledTripDate.defaultProps = {
  theme: _defaultTheme.default
};

var TripDate = function TripDate(_ref6) {
  var children = _ref6.children,
      dataTest = _ref6.dataTest,
      duration = _ref6.duration;
  return React.createElement(StyledTripDate, {
    "data-test": dataTest
  }, React.createElement(_Stack.default, {
    direction: "row",
    spacing: "condensed",
    shrink: true
  }, React.createElement(_Calendar.default, {
    size: "small"
  }), React.createElement(_Text.default, {
    type: "attention"
  }, children)), duration && React.createElement(_Badge.default, {
    icon: React.createElement(_Clock.default, null)
  }, duration));
};

var _default = TripDate;
exports.default = _default;
