// @kiwicom/orbit-components/lib/Button/consts.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TOKENS = exports.SIZE_OPTIONS = exports.TYPE_OPTIONS = void 0;
var TYPE_OPTIONS = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
  INFO: "info",
  SUCCESS: "success",
  WARNING: "warning",
  CRITICAL: "critical",
  FACEBOOK: "facebook",
  GOOGLE: "google",
  WHITE: "white"
};
exports.TYPE_OPTIONS = TYPE_OPTIONS;
var SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large"
};
exports.SIZE_OPTIONS = SIZE_OPTIONS;
var TOKENS = {
  // Size tokens
  heightButton: "heightButton",
  loadingWidth: "loadingWidth",
  loadingHeight: "loadingHeight",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  marginRightIcon: "marginRightIcon",
  // Type tokens
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonBordered: "backgroundButtonBordered",
  backgroundButtonBorderedHover: "backgroundButtonBorderedHover",
  backgroundButtonBorderedActive: "backgroundButtonBorderedActive",
  colorTextButton: "colorTextButton",
  colorTextButtonBordered: "colorTextButtonBordered",
  colorTextButtonHover: "colorTextButtonHover",
  colorTextButtonBorderedHover: "colorTextButtonBorderedHover",
  colorTextButtonActive: "colorTextButtonActive",
  colorTextButtonBorderedActive: "colorTextButtonBorderedActive",
  borderColorButton: "borderColorButton",
  borderColorButtonHover: "borderColorButtonHover",
  borderColorButtonActive: "borderColorButtonActive"
};
exports.TOKENS = TOKENS;
