// @kiwicom/orbit-components/lib/NotificationBadge/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Badge = _interopRequireWildcard(require("../Badge"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _consts = require("../Badge/consts");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledNotificationBadge = _styledComponents.default.div.withConfig({
  displayName: "NotificationBadge__StyledNotificationBadge",
  componentId: "zl2bxd-0"
})(["", "{width:", ";padding:0;}"], _Badge.StyledBadge, function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.widthBadgeCircled;
});

StyledNotificationBadge.defaultProps = {
  theme: _defaultTheme.default
};

var NotificationBadge = function NotificationBadge(props) {
  var _props$type = props.type,
      type = _props$type === void 0 ? _consts.TYPE_OPTIONS.NEUTRAL : _props$type,
      children = props.children,
      icon = props.icon,
      ariaLabel = props.ariaLabel,
      dataTest = props.dataTest;
  return React.createElement(StyledNotificationBadge, null, React.createElement(_Badge.default, {
    type: type,
    dataTest: dataTest,
    icon: icon,
    ariaLabel: ariaLabel
  }, !icon && children));
};

var _default = NotificationBadge;
exports.default = _default;
