// @kiwicom/orbit-components/lib/LazyImage/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledLazyImage = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var FORMATS = {
  WEBP: "webp",
  JPEG: "jpg"
};

var StyledLazyImage = _styledComponents.default.div.withConfig({
  displayName: "LazyImage__StyledLazyImage",
  componentId: "sc-1c5fd3o-0"
})(["width:100%;height:100%;position:absolute;"]);

exports.StyledLazyImage = StyledLazyImage;

var Image = _styledComponents.default.img.withConfig({
  displayName: "LazyImage__Image",
  componentId: "sc-1c5fd3o-1"
})(["z-index:", ";position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);min-width:100%;min-height:100%;opacity:", ";transition:opacity ", " ease-in-out;", ";"], function (_ref) {
  var visible = _ref.visible;
  return visible ? "1" : "0";
}, function (_ref2) {
  var visible = _ref2.visible;
  return visible ? "1" : "0";
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.durationFast;
}, function (_ref4) {
  var lowRes = _ref4.lowRes;
  return lowRes && (0, _styledComponents.css)(["filter:blur(3px);"]);
});

Image.defaultProps = {
  theme: _defaultTheme.default
};

var getPictureType = function getPictureType(picture) {
  var _TYPES;

  var TYPES = (_TYPES = {}, _defineProperty(_TYPES, FORMATS.WEBP, "image/webp"), _defineProperty(_TYPES, FORMATS.JPEG, "image/jpg"), _TYPES);
  return TYPES[picture];
};

var Picture = function Picture(_ref5) {
  var pictures = _ref5.pictures,
      name = _ref5.name,
      loaded = _ref5.loaded,
      onLoad = _ref5.onLoad,
      lowRes = _ref5.lowRes;
  return React.createElement("picture", null, Object.keys(pictures).map(function (picture) {
    return React.createElement(React.Fragment, {
      key: picture
    }, React.createElement("source", {
      srcSet: pictures[picture],
      type: getPictureType(picture)
    }), picture === "jpg" && React.createElement(Image, {
      onLoad: onLoad,
      src: pictures[picture],
      alt: name,
      visible: loaded,
      lowRes: lowRes
    }));
  }));
};

var LazyImage =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(LazyImage, _React$PureComponent);

  function LazyImage() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, LazyImage);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(LazyImage)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      loaded: false
    });

    _defineProperty(_assertThisInitialized(_this), "fullResLoaded", function () {
      _this.setState({
        loaded: true
      });
    });

    return _this;
  }

  _createClass(LazyImage, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          placeholder = _this$props.placeholder,
          original = _this$props.original,
          name = _this$props.name;
      var loaded = this.state.loaded;
      return React.createElement(StyledLazyImage, null, React.createElement(Picture, {
        pictures: original,
        name: name,
        loaded: loaded,
        onLoad: this.fullResLoaded
      }), placeholder && React.createElement(Picture, {
        pictures: placeholder,
        lowRes: true,
        name: name,
        loaded: !loaded
      }));
    }
  }]);

  return LazyImage;
}(React.PureComponent);

var _default = LazyImage;
exports.default = _default;
