// @kiwicom/orbit-components/lib/TripSector/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "TripDate", {
  enumerable: true,
  get: function get() {
    return _TripDate.default;
  }
});
Object.defineProperty(exports, "TripLayover", {
  enumerable: true,
  get: function get() {
    return _TripLayover.default;
  }
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _TripSegment = require("../TripSegment");

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _rtl = require("../utils/rtl");

var _TripDate = _interopRequireDefault(require("./TripDate"));

var _TripLayover = _interopRequireDefault(require("./TripLayover"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var StyledTripSector = _styledComponents.default.div.withConfig({
  displayName: "TripSector__StyledTripSector",
  componentId: "v14xkf-0"
})(["display:flex;position:relative;width:100%;padding-", ":2px;box-sizing:border-box;flex-direction:column;font-family:", ";&:before{content:\" \";display:block;position:absolute;background-color:", ";width:1px;height:100%;", ":7px;top:0;}", ":last-child{position:relative;", ":after{content:\" \";position:absolute;top:13px;right:0;bottom:0;left:0;width:16px;background:", ";}}"], _rtl.right, function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.fontFamily;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.paletteCloudNormal;
}, _rtl.left, _TripSegment.StyledTripSegment, _TripSegment.StyledTripSegmentMilestone, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.paletteWhite;
});

StyledTripSector.defaultProps = {
  theme: _defaultTheme.default
};

var TripSector = function TripSector(_ref4) {
  var children = _ref4.children,
      dataTest = _ref4.dataTest;
  return React.createElement(StyledTripSector, {
    "data-test": dataTest
  }, children);
};

var _default = TripSector;
exports.default = _default;
