// @kiwicom/orbit-components/lib/List/ListItem/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.IconContainer = exports.Item = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _CircleSmall = _interopRequireDefault(require("../../icons/CircleSmall"));

var _rtl = require("../../utils/rtl");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

var Item = _styledComponents.default.li.withConfig({
  displayName: "ListItem__Item",
  componentId: "sc-1vswvev-0"
})(["font-family:", ";display:flex;flex-direction:row;align-items:flex-start;margin-bottom:", ";&:last-child,&:last-of-type{margin:0;}"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.fontFamily;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.spaceXXSmall;
});

exports.Item = Item;
Item.defaultProps = {
  theme: _defaultTheme.default
};

var IconContainer = _styledComponents.default.div.withConfig({
  displayName: "ListItem__IconContainer",
  componentId: "sc-1vswvev-1"
})(["display:flex;align-items:center;margin:", ";"], function (_ref3) {
  var theme = _ref3.theme;
  return (0, _rtl.rtlSpacing)("0 ".concat(theme.orbit.spaceXSmall, " 0 0"));
});

exports.IconContainer = IconContainer;
IconContainer.defaultProps = {
  theme: _defaultTheme.default
};

var ListItem = function ListItem(_ref4) {
  var children = _ref4.children,
      _ref4$icon = _ref4.icon,
      icon = _ref4$icon === void 0 ? React.createElement(_CircleSmall.default, null) : _ref4$icon,
      dataTest = _ref4.dataTest;
  return React.createElement(Item, {
    "data-test": dataTest
  }, React.createElement(IconContainer, null, icon), children);
};

var _default = ListItem;
exports.default = _default;
