// @kiwicom/orbit-components/lib/Tooltip/helpers/isPosition.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isHorizontal = exports.isVertical = exports.isPositionTop = exports.isPositionBottom = exports.isPositionRight = exports.isPositionLeft = void 0;

var _consts = require("../consts");

var isPositionLeft = function isPositionLeft(p) {
  return p === _consts.POSITIONS.LEFT;
};

exports.isPositionLeft = isPositionLeft;

var isPositionRight = function isPositionRight(p) {
  return p === _consts.POSITIONS.RIGHT;
};

exports.isPositionRight = isPositionRight;

var isPositionBottom = function isPositionBottom(p) {
  return p === _consts.POSITIONS.BOTTOM;
};

exports.isPositionBottom = isPositionBottom;

var isPositionTop = function isPositionTop(p) {
  return p === _consts.POSITIONS.TOP;
};

exports.isPositionTop = isPositionTop;

var isVertical = function isVertical(p) {
  return isPositionTop(p) || isPositionBottom(p);
};

exports.isVertical = isVertical;

var isHorizontal = function isHorizontal(p) {
  return isPositionLeft(p) || isPositionRight(p);
};

exports.isHorizontal = isHorizontal;
