// @kiwicom/orbit-components/es/utils/Slide/index.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from "react";
import styled from "styled-components";
import defaultTheme from "../../defaultTheme";

const getMaxHeight = ({
  maxHeight
}) => {
  if (maxHeight === 0) return `0px`;
  if (!maxHeight) return undefined;
  return `${maxHeight}px`;
};

export const StyledSlide = styled.div.withConfig({
  displayName: "Slide__StyledSlide",
  componentId: "sc-1xbmdp2-0"
})(["max-height:", ";transition:max-height ", " linear;"], getMaxHeight, ({
  theme
}) => theme.orbit.durationFast);
StyledSlide.defaultProps = {
  theme: defaultTheme
};

class Slide extends React.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "state", {
      maxHeight: 0
    });

    _defineProperty(this, "expandTimeout", null);

    _defineProperty(this, "collapseTimeout", null);

    _defineProperty(this, "setMaxHeight", () => {
      const {
        maxHeight
      } = this.props;
      this.setState({
        maxHeight
      });
    });

    _defineProperty(this, "expandCallback", () => {
      this.setState({
        maxHeight: null
      });
    });

    _defineProperty(this, "collapseCallback", () => {
      this.setState({
        maxHeight: 0
      });

      if (this.expandTimeout && typeof clearTimeout === "function") {
        clearTimeout(this.expandTimeout);
        this.expandTimeout = null;
      }
    });
  }

  componentDidMount() {
    this.setMaxHeight();
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    if (snapshot) {
      if (this.props.expanded) {
        this.setMaxHeight();

        if (typeof setTimeout === "function") {
          this.expandTimeout = setTimeout(this.expandCallback, 150);
        }
      } else {
        if (this.state.maxHeight !== this.props.maxHeight) {
          this.setMaxHeight();
        }

        if (typeof setTimeout === "function") {
          this.collapseTimeout = setTimeout(this.collapseCallback, 1);
        }
      }
    }
  }

  componentWillUnmount() {
    if (typeof clearTimeout === "function") {
      if (this.expandTimeout) {
        clearTimeout(this.expandTimeout);
      }

      if (this.collapseTimeout) {
        clearTimeout(this.collapseTimeout);
      }
    }
  }

  getSnapshotBeforeUpdate(prevProps) {
    if (this.props.expanded === prevProps.expanded) return null;
    return true;
  }

  render() {
    const {
      children,
      expanded = false
    } = this.props;
    return React.createElement(StyledSlide, {
      maxHeight: this.state.maxHeight,
      expanded: expanded
    }, children);
  }

}

export default Slide;
